sklearn/covariance/_shrunk_covariance.py

```diff
@@ -23,6 +23,36 @@
 from ..utils._param_validation import Interval
 
 
+def _oas(X, *, assume_centered=False):
+    """Estimate covariance with the Oracle Approximating Shrinkage algorithm."""
+    # for only one feature, the result is the same whatever the shrinkage
+    if len(X.shape) == 2 and X.shape[1] == 1:
+        if not assume_centered:
+            X = X - X.mean()
+        return np.atleast_2d((X**2).mean()), 0.0
+    if X.ndim == 1:
```

Comment:
Since 1D arrays are deprecated, we should ideally remove this check as followed in the [PR](https://github.com/scikit-learn/scikit-learn/pull/24870). Please let me know if this assumption is wrong.